### PR TITLE
Added Animiru Support

### PR DESCRIPTION
**Pull request by juancolson**
Added Support for backup made using [Animiru](https://github.com/Quickdesh/Animiru) app, which is fork of aniyomi/tachiyomi/mihon that focuses on anime only. You can preview the change at [this link](https://animiru.netlify.app/).
Screenshot-2024-07-09-004142

### CODE REVIEW

File: site/scripts/library.js [hunk 1]
```diff
@@ -16,9 +16,19 @@ const re = RegExp('^https?://');
...
+ categories = window.data.backupAnimeCategory || [];
+   mangaItems = window.data.backupAnime;
+  } else {
+  categories = window.data.backupCategories || [];
+  mangaItems = window.data.backupManga;
+  }
+  
+  window.data.backupManga = mangaItems;
+  window.data.backupCategories = categories;
+
```
Comment (BrutuZ) [inline site/scripts/library.js:31]: Probably easier to rename the fields in the schema and avoid this extra logic

Comment (Death-Knight-552) [reply site/scripts/library.js]: I updated the code

File: site/scripts/loadBackup.js [hunk 2]
```diff
@@ -36,7 +38,7 @@ export function handleFileLoad(event, fork = 'mihon') {
                 enums: String,
                 bytes: String,
               });
-              initializeLibrary(); // Initialises Library with the Converter protobuf
+              initializeLibrary(fork); // Initialises Library with the Converter protobuf
```
Comment (Animeboynz) [inline site/scripts/loadBackup.js:41]:
```suggestion
              initializeLibrary(); // Initialises Library with the Converter protobuf
```

File: site/scripts/library.js [hunk 3]
```diff
@@ -16,10 +16,11 @@ const re = RegExp('^https?://');
 export { filterStatus, filterTracking, filterSource, sortOrder, activeTabId };
 
 // Function to Initialise the Tab Contents and Library from the JSON found in the data variable.
-export function initializeLibrary() {
-  const categories = window.data.backupCategories || [];
+export function initializeLibrary(fork = "mihon") {
```
Comment (Animeboynz) [inline site/scripts/library.js:19]:
```suggestion
export function initializeLibrary() {
```

File: site/schemas/schema-ani.proto [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    required string url = 2;
+    required string title = 3 [default = ""];
+    optional string artist = 4;
+    optional string author = 5;
+    optional string description = 6;
+    repeated string genre = 7;
+    optional int32 status = 8 [default = 0];
+    optional string thumbnailUrl = 9;
+    optional int64 dateAdded = 13 [default = 0];
+    repeated BackupEpisode episodes = 16;
```
Comment (Animeboynz) [inline site/schemas/schema-ani.proto:36]:
```suggestion
    repeated BackupEpisode chapters = 16;
```

**Review by Animeboynz — COMMENTED** [review]
Just so that the episodes are loaded too.
image

**Comment by Animeboynz** [comment]
Hey I apologize, but we have decided to keep this limited to [Mihon Endorsed Forks](https://mihon.app/forks/) only. You are free to maintain your own fork as long as you comply with the [existing licences](https://github.com/Animeboynz/Mihon-Backup-Viewer/blob/main/LICENSE) especially as you already have a hosted Netlify page. This is mainly because there are already dozens of forks and we have to draw the line somewhere.

**Pull request closed without merging**